browser/ui/webui/new_tab_page/brave_new_tab_message_handler.cc

```diff
@@ -689,6 +704,119 @@ void BraveNewTabMessageHandler::HandleTodayOnPromotedCardView(
   }
 }
 
+void BraveNewTabMessageHandler::HandleTodayGetDisplayAd(
+    const base::ListValue* args) {
+  AllowJavascript();
+  std::string callback_id;
+  args->GetString(0, &callback_id);
+  auto* ads_service_ = brave_ads::AdsServiceFactory::GetForProfile(profile_);
+  if (!ads_service_) {
+    ResolveJavascriptCallback(base::Value(callback_id),
+                              std::move(base::Value()));
+    return;
+  }
+  auto onAdReceived = base::BindOnce(
+      [](BraveNewTabMessageHandler* handler, std::string callback_id,
+         Profile* profile, const bool success, const std::string& dimensions,
+         const base::DictionaryValue& ad) {
+        if (!success) {
+          handler->ResolveJavascriptCallback(base::Value(callback_id),
+                                             std::move(base::Value()));
+          return;
+        }
+        // Mark as served since we're rendering immediately
+        auto* ads_service_ =
+            brave_ads::AdsServiceFactory::GetForProfile(profile);
+        std::string ad_id;
+        std::string creative_instance_id;
+        ad.GetString("uuid", &ad_id);
+        ad.GetString("creativeInstanceId", &creative_instance_id);
+        if (ads_service_ && !ad_id.empty() && !creative_instance_id.empty()) {
+          ads_service_->OnInlineContentAdEvent(
+              ad_id, creative_instance_id,
+              ads::mojom::BraveAdsInlineContentAdEventType::kServed);
```

Comment:
The ads library marks the ad as served

Reply:
ah ok I thought this was the change I was meant to make! That's great that we can do it like normal now that I'm only asking for the ad just before the unit is likely to be rendered. fixed.